app/models/review.rb

```diff
@@ -0,0 +1,49 @@
+class Review < ApplicationRecord
+  belongs_to :restaurant
+  after_create :attach_review_to_restaurant
+  after_destroy :remove_review_from_restaurant
+
+  def attach_review_to_restaurant
+    restaurant = Restaurant.find(restaurant_id)
```

Comment:
you don't have to manually select the Restaurant object by its id, 
because you defined `belongs_to :restaurant`, rails knows that `review.restaurant` is the object that returns from `restaurant_id ` and automatically does the select for you.
you can try this in the console and see the select rails performs.